core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceConnection.java

```diff
@@ -135,10 +141,17 @@
 	 * @param resolver
 	 */
 	protected SailSourceConnection(AbstractSail sail, SailStore store, FederatedServiceResolver resolver) {
+		this(sail, store, new SimpleEvaluationStrategyFactory(resolver), resolver);
```

Comment:
From my reading, EvaluationStrategy is something specific to the SailSourceConnection impl rather than the generic Sail API. It feels like there should be some layer between Sail and MemoryStore/NativeStore where EvaluationStrategy comes in.